tools/deserialize-proto/main.go

```diff
@@ -3,34 +3,33 @@ package main
 import (
 	"bufio"
 	"encoding/hex"
-	"encoding/json"
 	"fmt"
 	"log"
 	"os"
+	"reflect"
 	"strconv"
 	"strings"
 
+	"github.com/golang/protobuf/jsonpb"
 	"github.com/golang/protobuf/proto"
 	flag "github.com/spf13/pflag"
+	_ "github.com/stackrox/rox/generated/storage"
 )
 
 var protobufType *string = flag.String("type", "", "name of protobuf, e.g., storage.Alert")
 
-func unmarshalProto[T proto.Unmarshaler](t T, b []byte) error {
-	return t.Unmarshal(b)
-}
-
 func main() {
 	flag.Parse()
 
 	if *protobufType == "" {
 		log.Fatal("must provide --type")
 	}
 
-	messageType, ok := typeRegistry[*protobufType]
-	if !ok {
-		log.Fatalf("%s is an invalid storage type", *protobufType)
+	mt := proto.MessageType(*protobufType)
+	if mt == nil {
+		log.Fatalf("type %s could not be resolved to a protobuf message type", *protobufType)
 	}
+	msg := reflect.New(mt.Elem()).Interface().(proto.Message)
```

Comment:
Why not?
```suggestion
	var msg proto.Message
```

Reply:
Hmm tried things, it didn't work (the error is that you attempt to unmarshal a nil proto message, so apparently the registry isn't smart enough to do the init ourselves).